src/00b3_survey_mics/script.R

```diff
@@ -72,94 +88,144 @@ mics_paths <- load_sharepoint_data(
-         dataset = filetype, variable = id, 
-         var_raw = value, var_label_raw = label
-       ) %>% 
-       mutate(analysis = "circ", dataset = "mn") %>% # valid??
-       # don't have female date of births or interviews
-       filter(!var_raw %in% c("wdoi", "wdob")))
-) %>% 
+  (mics_indicators %>%
+     # change mics_indicator colnames to match variable_recode convention
+     rename(
+       # dataset = id, variable = filetype,
+       dataset = filetype, variable = id,
+       var_raw = value, var_label_raw = label
+     ) %>%
+     mutate(analysis = "circ", dataset = "mn") %>% # valid??
+     # don't have female date of births or interviews
+     filter(!var_raw %in% c("wdoi", "wdob")))
+  ) %>%
   # add in default row for mics_areas (hh7) for dataset == mn
   # also add rows for date of birth and date of interview (two different forms)
   bind_rows(
     tibble::tribble(
       ~survey_id,      ~dataset, ~variable,        ~var_raw,  ~var_label_raw,                ~analysis,
       "_default_mics", "mn",     "mics_area_name", "hh7",     "MICS area level",             "circ",
-      "ZWE2014MICS", "mn",     "dob",            "mwdob",   "Date of birth of man (CMC)",  "circ", 
-      "_default_mics", "mn",     "dob",            "mdob",    "Date of birth of man (CMC)",  "circ",
-      "ZWE2014MICS", "mn",     "doi",            "mwdoi",   "Date of interview men (CMC)", "circ",
-      "_default_mics", "mn",     "doi",            "mdoi",    "Date of interview men (CMC)", "circ"
+      "SWZ2010MICS",   "mn",     "dob",            "mdob",    "Date of birth of man (CMC)",  "circ",
+      "_default_mics", "mn",     "dob",            "mwdob",   "Date of birth of man (CMC)",  "circ",
+      "SWZ2010MICS",   "mn",     "doi",            "mdoi",    "Date of interview men (CMC)", "circ",
+      "_default_mics", "mn",     "doi",            "mwdoi",   "Date of interview men (CMC)", "circ",
     )
   )
 
+#' CAF2006MICS adjustments:
+#' * Dataset is "mm" (instead of "mn")
+#' * DOB variable is `wdob`
+#' * circ_status is `hha19`
+
+variable_recode <- variable_recode %>%
+  mutate(dataset = if_else(survey_id == "CAF2006MICS" & analysis == "circ", "mm", dataset)) %>%
+    bind_rows(
+    tibble::tribble(
+      ~survey_id,      ~dataset, ~variable,        ~var_raw,  ~var_label_raw,                ~analysis,
+      "CAF2006MICS",   "mm",     "circ_status",    "hha19",   "Etes vous circoncis",         "circ",
+      "CAF2006MICS",   "mm",     "dob",            "wdob",    "Date of birth (CMC)",         "circ",
+      "CAF2006MICS",   "mm",     "mnweight",       "meweight", "[Sampling weight]",          "circ",
+      "CAF2006MICS",   "mm",     "cluster_id",     "hh1",     "Cluster ID - man dataset",      NA,
+      "CAF2006MICS",   "mm",     "household",      "hh2",     "Household ID - man dataset",    NA,
+      "CAF2006MICS",   "mm",     "line",           "ln",      "Line number - man dataset",     NA
+      )
+  )
```

Comment:
To confirm I have understood the system correctly, eventually all of this should be updated in the master .XLSX right?

Fine to keep here for now, but just want to make sure I understand the intended workflow.

Reply:
Yep, that's correct. Why did you remove the ZWE2014MICS stuff?

Reply:
And also change the _default_mics var_raw (i.e. mwdob -> mdob)?

Reply:
I found that `ZWE2014MICS` followed the standard pattern, it was `SWZ2010MICS` that had customised variable names from the rest of the surveys.

So I (1) changed the `_default_mics` to match the variable names used in the ZWE2014MICS (and rest of the MICS), and (2) changed `SWZ2010MICS` to have the custom pattern.